pdns/bpf-filter.cc

```diff
@@ -320,6 +321,23 @@ BPFFilter::BPFFilter(std::unordered_map<std::string, MapConfiguration>& configs,
     throw std::runtime_error("Unsupported eBPF map format, the current internal implemenation only supports the legacy format");
   }
 
+  struct rlimit old_limit;
+
+  if ( getrlimit(RLIMIT_MEMLOCK, &old_limit) != 0) {
+    throw std::runtime_error("Unable to get memory lock limit: " + stringerror());
+  }
+
+  /* Check if the current soft memlock limit is 64k */ 
+  if (old_limit.rlim_cur < 67108865) {
+    struct rlimit new_limit;
```

Comment:
I think it would be nice to log at info level what we are about to do, something like:
```c++
const size_t newLimit = 1073741824;
infolog("The current limit of locked memory (soft: %d, hard: %d) is too low for eBPF, trying to raise it to %d", old_limit.rlim_cur, old_limit.rlim_max, newLimit);
```

Reply:
I have added the info log rather than throwing an exception